### DISCUSSION TITLE
Native CSV File import

### DISCUSSION THREAD

**Discussion by stpruss** [ExpressionEngine/ExpressionEngine#2984] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
EE lacks a good, configurable data import feature (ala DataGrab). The current third-party option appears to be unsupported

**Describe the solution you'd like**
A tool that allows one to select a file via file upload or URL and configure the columns in it to import into a specified EE channel and specific fields. Import report would be good as well to see which imports passed/failed

**Describe alternatives you've considered**
DataGrab is the only option, but it has gotten slow and does not seem to be supported anymore by the Dev.

**Teachability, Documentation, Adoption, Migration Strategy**
Would just need standard documentation to explain it's use.

- [ ] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by bsrweb** [comment]
+1

And I'll just repeat what I commented on Slack: this seems like a critical piece of functionality to help grow EE's marketshare to allow developers to more easily migrate non-EE sites to EE.

**Comment by blue-dreamer** [comment]
+1 here as well.

If we are to attract new developers to EE the ability to import data may be a key factor. EE has had  a member import facility for many years so why not a native method of importing entry data?

**Comment by blairliikala** [comment]
A reoccurring/scheduled method for import would be critical for me.

**Comment by calanh** [comment]
+1

**Comment by firebrandmedia** [comment]
+1

**Comment by carolinecblaker** [comment]
+1 Sites that rely on sync with external systems (like SAP) will likely go to other web software over giving up their integrations, without the ability to access this feature. For my own part I don't care if it comes in third party or native.

**Comment by ssssct** [comment]
Any improvement on imports should allow to import members as well. EE's native member import utility is far from perfect. It's not very user friendly, completely unusable for clients, does not allow automatic cron imports and the import is cancelled if anything — really anything — seems to go wrong. Preventing to import large amounts of data and making it almost useless.
